contracts/modules/royalty/policies/IpRoyaltyVault.sol

```diff
@@ -0,0 +1,237 @@
+
+    /// @notice LAP royalty policy address
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    IRoyaltyPolicyLAP public immutable ROYALTY_POLICY_LAP;
+
+    /// @notice Ip id to whom this royalty vault belongs to
+    address public ipId;
+
+    /// @notice Amount of unclaimed royalty tokens
+    uint32 public unclaimedRoyaltyTokens;
+
+    /// @notice Last snapshotted timestamp
+    uint256 public lastSnapshotTimestamp;
+
+    /// @notice Amount of revenue token in the ancestors vault
+    mapping(address token => uint256 amount) public ancestorsVaultAmount;
+
+    /// @notice Indicates if a given ancestor address has already claimed
+    mapping(address ancestorIpId => bool) public isClaimedByAncestor;
+
+    /// @notice Amount of revenue token in the claim vault
+    mapping(address token => uint256 amount) public claimVaultAmount;
+
+    /// @notice Amount of tokens claimable at a given snapshot
+    mapping(uint256 snapshotId => mapping(address token => uint256 amount)) public claimableAtSnapshot;
+
+    /// @notice Amount of unclaimed tokens at the snapshot
+    mapping(uint256 snapshotId => uint32 tokenAmount) public unclaimedAtSnapshot;
+
+    /// @notice Indicates whether the claimer has claimed the revenue tokens at a given snapshot
+    mapping(uint256 snapshotId => mapping(address claimer => mapping(address token => bool)))
+        public isClaimedAtSnapshot;
+
+    /// @notice Royalty tokens of the vault
+    EnumerableSet.AddressSet private _tokens;
+
+    /// @notice Constructor
+    /// @param royaltyPolicyLAP The address of the royalty policy LAP
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(address royaltyPolicyLAP) {
+        if (royaltyPolicyLAP == address(0)) revert Errors.IpRoyaltyVault__ZeroRoyaltyPolicyLAP();
+        ROYALTY_POLICY_LAP = IRoyaltyPolicyLAP(royaltyPolicyLAP);
+        _disableInitializers();
+    }
+
+    // TODO: adjust/review for upgradeability
+    /// @notice Initializer for this implementation contract
+    /// @param name The name of the royalty token
+    /// @param symbol The symbol of the royalty token
+    /// @param supply The total supply of the royalty token
+    /// @param unclaimedTokens The amount of unclaimed royalty tokens reserved for ancestors
+    /// @param ipIdAddress The ip id the royalty vault belongs to
+    function initialize(
+        string memory name,
+        string memory symbol,
+        uint32 supply,
+        uint32 unclaimedTokens,
+        address ipIdAddress
+    ) external initializer {
+        if (ipIdAddress == address(0)) revert Errors.IpRoyaltyVault__ZeroIpId();
+
+        ipId = ipIdAddress;
+        lastSnapshotTimestamp = block.timestamp;
+        unclaimedRoyaltyTokens = unclaimedTokens;
+
+        _mint(address(this), unclaimedTokens);
+        _mint(ipIdAddress, supply - unclaimedTokens);
+
+        __ReentrancyGuard_init();
+        __ERC20Snapshot_init();
+        __ERC20_init(name, symbol);
```

Comment:
I suggest we have a common prefix of the royalty token, both name and symbol

Reply:
We have "Royalty Token" as name and "RT" as symbol for all royalty tokens

Reply:
If that's the case, should we just hardcode it?

Reply:
Was keeping the IpRoyaltyVault less opinionated and just pass in. We can though - lmk.